scalafix-core/shared/src/main/scala/scalafix/internal/rule/RemoveUnusedTerms.scala

```diff
@@ -0,0 +1,42 @@
+package scalafix.internal.rule
+
+import scala.meta._
+import scalafix.Patch
+import scalafix.SemanticdbIndex
+import scalafix.rule.RuleCtx
+import scalafix.rule.SemanticRule
+
+case class RemoveUnusedTerms(index: SemanticdbIndex)
+    extends SemanticRule(index, "RemoveUnusedTerms") {
+
+  private val unusedTerms = {
+    val UnusedLocalVal = """local (.*) is never used""".r
+    index.messages.toIterator.collect {
+      case Message(pos, _, UnusedLocalVal(_*)) =>
+        pos
+    }.toSet
+  }
+
+  private def isUnused(defn: Defn) =
+    unusedTerms.contains(defn.pos)
+
+  private def removeDeclarationTokens(i: Defn, rhs: Term): Tokens = {
+    val startDef = i.tokens.start
```

Comment:
TIL I have not noticed Tokens.start 😄

Reply:
Yeah, I struggled a bit with this function as `Tokens.filter` returns a `IndexSeq[Token]` rather than a `Tokens`. Also, the constructor of `Tokens` is private to the package `meta`, so writing something like 
```
i.tokens.filterNot(rhs.tokens.contains)
```
 would have been nicer but unfortunately it didn't quite work :(